bootstrap.sh

```diff
@@ -0,0 +1,133 @@
+#! /usr/bin/env bash
+set -e
+set -o pipefail
+
+# Sanity checks
+sanity(){
+    # Ensure the script is running on macOS
+    if [ $(uname -s) != "Darwin" ]; then
+	echo "This script is for use with macOS!"
+	exit 1
+    fi
+
+    # Ensure script is not being run with root privileges
+    if [ $EUID -eq 0 ]; then
+	echo "Please don't run this script with root privileges!"
+	exit 1
+    fi
+
+    # Ensure Nix has already been installed
+    if [[ ! $(type nix-env 2>/dev/null) ]]; then
+	echo -e "Cannot find "$YELLOW"nix-env"$ESC" in the PATH"
+	echo "Please ensure you've installed and configured your Nix environment correctly"
+	echo -e "Install instructions can be found at: "$BLUE_UL"https://nixos.org/nix/manual/#chap-quick-start"$ESC""
+	exit 1
+    fi
+
+    # Check if nix-darwin is already present
+    if [[ $(type darwin-rebuild 2>/dev/null) ]]; then
+	echo -e "It looks like "$YELLOW"nix-darwin"$ESC" is already installed..."
+	exit 1
+    fi
+}
+
+# Exit with message
+exit_message(){
+    echo -e ""$RED"$1"$ESC"" >&2
+    exit 1
+}
+
+# Installer
+install(){
+    echo -e ""$BLUE"Welcome to the nix-darwin installer!"$ESC""
+
+    # Prompt for nix package upgrade
+    echo -e "Ensuring "$YELLOW"nixpkgs"$ESC" version meets requirements..."
+    echo -e "To do this, the following will be run to upgrade the "$YELLOW"nix"$ESC" package:"
+    echo -e ""$YELLOW"nix-env -iA nixpkgs.nix"$ESC"\n"
+    echo "If you have a recent install of Nix, this may not be necessary, but should not cause any harm to run"
+    while true ; do
+	read -p "Would you like to upgrade? [y/n] " ANSWER
+	case $ANSWER in
+	    y|Y)
+		echo "Proceeding with upgrade..."
+		nix-env -iA nixpkgs.nix
+		break
+		;;
+	    n|N)
+		echo "Proceeding without upgrading..."
+		break
+		;;
+	    *)
+		echo "Please answer 'y' or 'n'..."
+	esac
+    done
+
+    # Prompt for initial sudo password & keep alive
+    echo "Please enter your password for sudo authentication"
+    sudo -k
+    sudo echo "sudo authenticaion successful!"
+    while true ; do sudo -n true ; sleep 60 ; kill -0 "$$" || exit ; done 2>/dev/null &
+
+
+    # Link run directory
+    echo "Setting up /run..."
+    test -L /run || sudo ln -s /private/var/run /run
+
+    # Fetch the nix-darwin repo as a zip (shouldn't assume presence of git)
+    REPO_DOWNLOAD=/tmp/nix-darwin-$(date +%Y%m%d).zip
+    if [ ! -f $REPO_DOWNLOAD ]; then
+	echo -e ""$YELLOW"Fetching nix-darwin repo..."$ESC""
+	curl -Lo $REPO_DOWNLOAD https://github.com/LnL7/nix-darwin/archive/master.zip || \
```

Comment:
Sorry, I'm not sure I follow - I'm pretty new to Nix, so you'll have to explain this one